### PR TITLE
JAMES-3168 TestJMAPServerModule: remove arbitrary default search limit

### PR DESCRIPTION
**Pull request by chibenwa**
Only tests that explicitly require it need to supply this value via separate
guice module.

Others can simply rely on application defaults.

### CODE REVIEW

**Comment by chibenwa** [comment]
MethodsModule defined the defaulmt to 256

```
bindConstant().annotatedWith(Names.named(GetMessageListMethod.MAXIMUM_LIMIT)).to(GetMessageListMethod.DEFAULT_MAXIMUM_LIMIT);
```

File: server/container/guice/cassandra-guice/src/test/java/org/apache/james/AuthenticatedCassandraJamesServerTest.java [hunk 1]
```diff
@@ -50,7 +50,7 @@
             .server(configuration -> GuiceJamesServer.forConfiguration(configuration)
                 .combineWith(ALL_BUT_JMX_CASSANDRA_MODULE)
                 .overrideWith(binder -> binder.bind(TextExtractor.class).to(PDFTextExtractor.class))
-                .overrideWith(TestJMAPServerModule.limitToTenMessages())
+                .overrideWith(new TestJMAPServerModule())
```
Comment (mbaechler) [inline server/container/guice/cassandra-guice/src/test/java/org/apache/james/AuthenticatedCassandraJamesServerTest.java:53]: what about `TestJMAPServerModule.default()`?

Comment (chibenwa) [reply server/container/guice/cassandra-guice/src/test/java/org/apache/james/AuthenticatedCassandraJamesServerTest.java:53]: What's the added value of `default()` if there is no other possible value?

**Comment by Arsnael** [comment]
Merged

**Pull request closed without merging**